src/java.base/share/classes/com/sun/crypto/provider/PBKDF2KeyImpl.java

```diff
@@ -247,33 +247,29 @@ public String getFormat() {
      * Calculates a hash code value for the object.
      * Objects that are equal will also have the same hashcode.
      */
+    @Override
     public int hashCode() {
         try {
-            int retval = 0;
-            for (int i = 1; i < this.key.length; i++) {
-                retval += this.key[i] * i;
-            }
-            return (retval ^= getAlgorithm().toLowerCase
-                    (Locale.ENGLISH).hashCode());
+            return Arrays.hashCode(this.key)
+                    ^ getAlgorithm().toLowerCase(Locale.ENGLISH).hashCode();
         } finally {
             // prevent this from being cleaned for the above block
             Reference.reachabilityFence(this);
         }
     }
 
+    @Override
     public boolean equals(Object obj) {
         try {
             if (obj == this) {
                 return true;
             }
 
-            if (!(obj instanceof SecretKey)) {
+            if (!(obj instanceof SecretKey that)) {
                 return false;
             }
 
-            SecretKey that = (SecretKey) obj;
-
-            if (!(that.getAlgorithm().equalsIgnoreCase(getAlgorithm()))) {
+            if (!(that.getAlgorithm().equalsIgnoreCase(getAlgorithm()))){
```

Comment:
why removing the space before the "{"?

Reply:
It surprises me as much as it probably surprises you. Looks like a sloppy merge (821f617ffae) on my part.

Reply:
I retraced my steps to see how the merge introduced that artefact. This might be useful to others too.

The original merge was done via an IDE, which has this (usually) helpful 1-click feature called "Resolve Simple Conflicts". It's that feature that ate that whitespace character. To its credit, the rest of the resolution seems to have been done correctly. When I examined the original merge before pushing it to the PR, I didn't spot that whitespace; my bad. I'll reintroduce it.

Thanks for your careful review.